src/core/idle.cpp

```diff
@@ -0,0 +1,39 @@
+#include <wayfire/idle.hpp>
+#include <wayfire/util/log.hpp>
+#include "core-impl.hpp"
+
+static inline void set_inhibited(bool state)
+{
+    /* NOTE: inhibited -> NOT enabled */
+    wlr_idle_set_enabled(wf::get_core().protocols.idle, NULL, !state);
+}
+
+wf::idle_inhibitor_t::idle_inhibitor_t()
+{
+    LOGD("creating idle inhibitor %p, previous count: %d", this,
+        wf::get_core_impl().idle_inhibitors);
+    if (++wf::get_core_impl().idle_inhibitors == 1)
```

Comment:
seems like this should be a public method of core_impl_t

Reply:
The increment/decrement operation?

Reply:
Yes, which also calls enable/disable on the wlr_idle

Reply:
hm, my thinking was that I should avoid adding methods to the core_impl because these operations should only ever be done from this class, not from anywhere else

Reply:
then make the counter a static member variable? My main concern is that "friend class" is rarely a good idea, and here it seems like the counter belongs to the idle inhibitor class.

Reply:
oh yeah that's a good solution